src/test/java/org/broadinstitute/hellbender/utils/io/IOUtilsUnitTest.java

```diff
@@ -41,6 +44,212 @@ public void tearDown() {
         }
     }
 
+    private void assertContentsTheSame(final Path baseActualPath, final Path baseExpectedPath) {
+        try {
+            // Check that the files and directories are the same:
+            Files.find(baseActualPath, Integer.MAX_VALUE,
+                    (actualPath, fileAttributes) -> {
+
+                        // First check that the corresponding file exists in our expected unzipped archive:
+                        final Path expectedPath = baseExpectedPath.resolve(baseActualPath.relativize(actualPath));
+
+                        final Path actualFileName = actualPath.getFileName();
+                        final Path expectedFileName = expectedPath.getFileName();
+
+                        Assert.assertEquals(actualFileName, expectedFileName);
+
+                        // For regular files, make sure the contents of the corresponding files are the same:
+                        if ( fileAttributes.isRegularFile() && !fileAttributes.isSymbolicLink()) {
+                            try {
+                                final byte[] actualContents   = Files.readAllBytes(actualPath);
+                                final byte[] expectedContents = Files.readAllBytes(expectedPath);
+
+                                Assert.assertEquals(actualContents, expectedContents);
+                            }
+                            catch (final IOException ex) {
+                                throw new GATKException("Could not verify contents of files: " + actualPath.toUri() + " AND " + expectedPath.toUri(), ex);
+                            }
+                        }
+
+                        return true;
+                    }
+            );
+        }
+        catch (final IOException ex) {
+            throw new GATKException("Could not verify identical contents of test directories!");
+        }
+    }
+
+    @DataProvider
+    private Object[][] provideForTestExtractTarGz() {
+        return new Object[][] {
+                { null },
+                { createTempDir("provideForTestExtractTarGz_Case2").toPath() },
+        };
+    }
+
+    @Test(dataProvider = "provideForTestExtractTarGz")
+    public void testExtractTarGz(final Path destDir) {
+
+        final Path outputDataSourcesPath;
+
+        // Prepare our output directory:
+        if ( destDir == null ) {
+            final File tmpDir = createTempDir("IOUtilsUnitTest_testExtractTarGz");
+            tmpDir.deleteOnExit();
```

Comment:
`createTempDir()` calls `deleteRecursivelyOnExit()` for you, so the manual `deleteOnExit()` calls are not needed (applies to the other test cases below as well).

Reply:
I know, but for whatever reason I found that it did not delete the directories at least once when I was testing.  This seemed to fix it.

Reply:
I can take it out and try again.

Reply:
Fixed.